R/transform_.R

```diff
@@ -0,0 +1,94 @@
+#' Transform steepness into logit space
+#'
+#' Transform the steepness parameter that can range from 0.2 to 1.0 into logit
+#' space for use in a FIMS model. This is useful for ensuring that the steepness
+#' parameter is bounded within the desired range.
+#'
+#' @param h A vector of numeric steepness parameter values that must be
+#'   between 0.2 and 1.0.
+#' @export
+#' @return
+#' A vector of numeric values representing the transformed steepness parameters
+#' in logit space.
+#' @examples
+#' # Transform steepness value of 0.5
+#' transform_steepness(0.5)
+transform_steepness <- function(h) {
+  if (h < 0.2 || h > 1.0) {
+    cli::cli_abort(
+      c(
+        "x" = "Steepness (h) must be a numeric value between 0.2 and 1.0.",
+        "i" = "Provided value{?s}: {h}"
+      )
+    )
+  }
+  # function to convert steepness into required parameter space
+  transform_logistic(h, min = 0.2, max = 1.0)
+}
+
+#' Transform a value using the logistic transformation
+#'
+#' The logistic transformation allows users to transform a value with bounds
+#' and place it in infinite space to allow for the use of a normal curve.
+#'
+#' @details
+#' The logistic transformation is largely used to transform numbers from the
+#' [0,1] scale to [-Inf,Inf] but within FIMS we also use it to transform
+#' numbers between two known bounds to infinite bounds. An example is with
+#' the steepness parameter that is bound between [0.2,1.0]. The logistic
+#' transformation is defined as \eqn{logistic(x) = ln(x/(1-x))} for
+#' \eqn{x in (0,1)}. Where we allow users to define the bounds
+#' \eqn{logistic(x) = -ln(max - x) + ln(x - min)}.
+#' @param x A vector of real values.
+#' @param min,max A single value or a vector the same length as `value` that
+#'   specifies the minimum/maximum value(s) allowed in the untransformed space.
+#' @export
+#' @return
+#' A vector of transformed values the same length as `values`.
+#' @seealso
+#' * [transform_steepness()]
+#' @examples
+#' # Transform the steepness parameter, where the latter uses the former
+#' # internally, and thus, the answers will be the exact same but using the
+#' # latter allows for less code and assurance that the bounds are correct.
+#' transform_logistic(0.3, 0.2, 1.0)
+#' transform_steepness(0.3)
+#'
+#' # transform a vector
+#' transform_logistic(1:10, min = 0, max = 10)
+#' transform_logistic(1:10, min = 0, max = 11:20)
+transform_logistic <- function(x, min, max) {
```

Comment:
I think it is more straightforward to call this the logit than the transform_logistic. That matches our naming scheme for the C++ equivalent [here](https://github.com/NOAA-FIMS/FIMS/blob/eca8095469b205dd9684294e70ce8eda56afa3ba/inst/include/common/fims_math.hpp#L160).